provider/sidecar.go

```diff
@@ -92,11 +106,8 @@ func (provider *Sidecar) Provide(configurationChan chan<- types.ConfigMessage, p
 	if err != nil {
 		log.Fatalln("Error reloading Sidecar config", err)
 	}
-	err = provider.loadSidecarConfig(states.ByService())
-	if err != nil {
-		return err
-	}
-	return nil
+
+	return provider.loadSidecarConfig(states.ByService())
```

Comment:
This might not work out easily, because the Sidecar `watch` endpoint returns `state.ByService()`, which is a `map[string][]*service.Service`.

Given this, you mentioned submitting a PR to Sidecar to add a parameter to this endpoint that allows the consumer to specify whether they want to get the state "ByService" or not. I'll proceed to make the required changes in Sidecar for this to work.